constitution.md

```diff
@@ -0,0 +1,232 @@
+* red lighting/green lighting releases
+
+Developers who have been inactive for six months or more will be removed from the group.
+
+A developer interested in becoming a reviewer should have a wide body of commits and begin reviewing submitted pull 
+requests to practice and build up a body of reviews, both to demonstrate their expertise but also to increase their 
+familiarity with the repository. These pull-requests will also require reviews from a Reviewer. Once a developer has 
+had over 10 reviews on a project, they can submit a request to the administrators to become a Reviewer for that project.
+
+Developers can contribute feedback by testing scheduled releases and fixes, and these members are critical to ensuring 
+Chisel's backwards compatibility, as well as discovering bugs before a release. 
+
+### Volunteers
+Volunteers are people who want to contribute to a Chisel project, but do not commit to a regular contribution. They do 
+not have voting rights, and no registration is required. Volunteers may attend development meetings. Volunteers may 
+contribute through PR’s to Chisel repositories, and are encouraged to review and comment but they cannot accept PR requests. It is hoped that volunteers will become developers.
+
+## Collaboration Policy
+CWG’s standard communication channels and development policy is outlined as follows, and will likely be adopted for other 
+associated CHIPS Alliance working groups.
+
+## Communication Channels
+High level leadership directives will be published via mailing list, and minutes of these discussions will be posted on the 
+Chisel website.
+
+All technical development discussions will be done either on Github with Github issues, pull requests, and comments, or via 
+a recurring open development meeting with video conferencing. Open development meetings will be run by project 
+administrators or senior reviewers.
+
+User-facing support is primarily through answering questions on **Stack Overflow**, contributing to and maintaining a **Github 
+Wiki page**, and promoting community dialogue with an associated **Gitter** channels. The purpose of Gitter is to move discussion 
+from mailing lists to a more modern communication outlet.
+
+
+### Development Policy
+Generally, development on mature products will release minor/major versions, where minor versions are backwards compatible, 
+and major versions can break ok API’s. Old API’s will be deprecated prior to removal.
+
+New features which are experimental could be released in an **experimental** package, whereby all API’s in this package are 
+subject to API changes without deprecation, and may change on a minor release (although this will be avoided if possible). 
+Successful experimental features/APIs will eventually be migrated to other packages.
+
+All code contributions to mature projects (Chisel, FIRRTL, etc.) must undergo a review process and pass continuous 
+integration tests prior to merging into the master branch.
```

Comment:
While I am a 100% CI superfan, sometimes CI does introduce issues (gets hung up, some yosys check fails for silly reasons, etc). Maybe we can add a clause (if it's not already elsewhere) to add a backup provision for this?

Reply:
Maybe just "pass testing". We already have the ability to merge PRs that rightly fail tests (`[skip chisel tests]` in FIRRTL). Also, this does technically enshrine `master` branch into the constitution which implies a specific development model. Maybe we can generalize this?

I think, broadly, this is trying to say that:

1. We have a culture of testing
1. Projects should have reviewers
1. People should not be committing directly to a release branch
1. These rules *may* not apply if you have a very immature project

Reply:
Replace with 'pass continuous integration tests' with 'rigorous testing' ?